test/games/am2r/gui/preset_settings/test_preset_am2r_goal.py

```diff
@@ -65,3 +65,63 @@ def test_preferred_dna(
 
     assert editor.configuration.artifacts.required_artifacts == expected_artifacts
     assert tab.dna_slider.value() == expected_artifacts
+
+
+def test_restricted_placement(
+    skip_qtbot,
+    am2r_game_description,
+    preset_manager,
+):
+    # Setup
+    game = am2r_game_description.game
+    base = preset_manager.default_preset_for_game(game).get_preset()
+    preset = dataclasses.replace(base, uuid=uuid.UUID("b41fde84-1f57-4b79-8cd6-3e5a78077fa6"))
+    base_configuration = preset.configuration
+    options = MagicMock()
+    assert isinstance(base_configuration, AM2RConfiguration)
+
+    tab = PresetAM2RGoal(editor := PresetEditor(preset, options), am2r_game_description, MagicMock())
+    skip_qtbot.addWidget(tab)
+    tab.on_preset_changed(preset)
+    artifact_count = editor.configuration.artifacts.required_artifacts
```

Comment:
mypy error.
You might want to move the assert from below somewhere further to the top.